### PR TITLE
Figure speed regulation + logs support

### PR DESCRIPTION
**Pull request by s373r**

### CODE REVIEW

File: src/engine_event_handler.py [hunk 1]
```diff
@@ -24,48 +30,83 @@ def __use_next_figure(self) -> None:
...
+        new_points, new_position = FigureMover.move_left(self.__current_figure)
+
+        if self._board.is_cells_empty(new_points):
+            self.__current_figure.position = new_position
+
+    def __handle_figure_move_down(self) -> None:
+        new_points, new_position = FigureMover.move_down(self.__current_figure)
+
+        if self._board.is_cells_empty(new_points):
+            self.__current_figure.position = new_position
```
Comment (alena-bartosh) [inline src/engine_event_handler.py:62]: Looks like duplicates. Do you like my previous variant with `event->moving_func` mapping?

File: src/engine_event_handler.py [hunk 2]
```diff
@@ -24,48 +30,83 @@ def __use_next_figure(self) -> None:
...
+            return
+
+        self._board.set_cells_with_value(self.__current_figure.get_points(), True)
+
+        removed_lines = self._board.remove_lines_if_needed()
+
+        if removed_lines > 0:
+            self.__decrease_ticks_for_move_down(removed_lines)
+
+            self.__score_holder.increment(removed_lines)
```
Comment (alena-bartosh) [inline src/engine_event_handler.py:73]:
```suggestion
            self.__decrease_ticks_for_move_down(removed_lines)
            self.__score_holder.increment(removed_lines)
```

File: src/engine_event_handler.py [hunk 3]
```diff
@@ -24,48 +30,83 @@ def __use_next_figure(self) -> None:
...
+
```
Comment (alena-bartosh) [inline src/engine_event_handler.py:74]:
```suggestion
```

File: src/main.py [hunk 4]
```diff
@@ -7,8 +7,10 @@
 
 def main(stdscr) -> None:
     board = Board(height=20, weight=10)
-    event_handler = EngineEventHandler(stdscr, board)
-    engine = Engine(stdscr, event_handler, tick_duration_in_sec=1)
+    event_handler = EngineEventHandler(stdscr,
+                                       board,
+                                       initial_ticks_for_move_down=50)
+    engine = Engine(stdscr, event_handler, tick_duration_in_sec=0.02)
```
Comment (alena-bartosh) [inline src/main.py:13]: Nice idea

File: src/renderer.py [hunk 5]
```diff
@@ -80,11 +80,24 @@ def __draw_statistic(self, board: Board, score_holder: ScoreHolder) -> None:
 
         self.__draw_string(x, y, str(score_holder.get_burned_line_count()))
 
+    def __show_logs(self, log_lines: list[str]) -> None:
+        y = 0
+        x = 25
+
+        self.__draw_string(y, x, 'Logs:')
+
+        for i, log_line in enumerate(log_lines[-20:]):
```
Comment (alena-bartosh) [inline src/renderer.py:89]:
```suggestion
        for i, log_line in enumerate(log_lines[-20:]):  # get latest 20 elements from logs
```

File: src/engine_event_handler.py (hunk 1 above)
Comment (s373r) [reply src/engine_event_handler.py:62]: Yes, for sure -- I gonna implement your approach, any concerns?

**Review by alena-bartosh — APPROVED** [review]

**Pull request merged**